modules/mobile/bin/health_checks/endpoint_tester.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+require 'colorize'
+require 'thor'
+require 'yaml'
+require_relative 'token_fetcher'
+require_relative 'client'
+
+class EndpointTester < Thor
+  TEST_DATA_DIR = File.join(File.dirname(__FILE__), 'request_data')
+
+  desc 'run tests', 'tests endpoints based on yaml inputs'
+  option :base_url
+  option :test_name
+  def run_tests
+    @users = {}
+    @results = {}
+    files = Dir["#{TEST_DATA_DIR}/**/*.yaml"]
+    files.each do |f|
+      test_data = YAML.safe_load(File.read(f))
+      sequence = test_data['sequence']
+      sequence.each do |test_case|
+        test_data = test_case['case']
+        next if options[:test_name] && test_data['name'] != options[:test_name]
+
+        run_individual_test_case(test_data)
+      end
+    end
+    process_results
+  end
+
+  private
+
+  def run_individual_test_case(data)
+    method = data['method']
+    path = data['request']['path']
+    user_name = data['request']['user']
+    client = client(user_name)
+
+    response = case method
+               when 'GET'
+                 client.get(path)
+               else
+                 abort("Invalid method: #{method}".red)
+               end
+
+    validate_response(data, response)
+  end
+
+  def validate_response(expected_data, response)
+    test_name = expected_data['name']
+    @results[test_name] = []
+    @current_test = @results[test_name]
+
+    expected_status = expected_data.dig('response', 'status')
+    correct_status_received = expected_status == response.status
+
+    if correct_status_received
+      received_data = JSON.parse(response.body)['data']
+
+      count = expected_data.dig('response', 'count')
+      add_error('count', count, received_data.count) if count && count != received_data.count
+
+      expected_data = expected_data.dig('response', 'data')
+      # ensuring that both data sets are hashes to avoid having a special case when data is an array
```

Comment:
I don't love that I have to convert these to objects, but it makes the `compare_data` method wants to have a key to use as a descriptor. I considered other options but nothing seemed any better.